pkg/route/registry/route/strategy.go

```diff
@@ -45,7 +45,10 @@ func (s routeStrategy) PrepareForCreate(obj runtime.Object) {
 	route.Status = api.RouteStatus{}
 	// Limit to kind/name
 	// TODO: convert to LocalObjectReference
-	route.Spec.To = kapi.ObjectReference{Kind: route.Spec.To.Kind, Name: route.Spec.To.Name}
+	route.Spec.To = api.RouteTargetReference{Kind: route.Spec.To.Kind, Name: route.Spec.To.Name, Weight: 1}
```

Comment:
This block is no longer necessary - you can remove it and the block below because we no longer have fields to strip.